### PR TITLE
fix: add lastActiveLogIndex

### LINKED ISSUE #4872: Add position indicator in session logs
https://github.com/highlight/highlight/pull/6303 added a visual indicator for errors and network resources, and we'd like to add a similar indicator for console logs.

### PR DESCRIPTION
**Pull request by jainpawan21**
## Summary
1. Add `lastActiveLogIndex` and active styling

/claim #4872 
closes #4872

### CODE REVIEW

**Review by et — CHANGES_REQUESTED** [review]
This is looking great. I did notice that we already have logic to track the current log index (see `foundIndex`). I think what you have here (using `findLastActiveEventIndex`) makes a lot of sense but it would be great to drop the old logic.

Can we try the following:

* Remove the `foundIndex` code
* Remove the `useEffect` with the comment `// Logic for scrolling to current entry.`
* Update references to `foundIndex` to point to `lastActiveLogIndex`
* Update the `current` prop to `<MessageRow />` to be `_index == lastActiveLogIndex`
* Add a new prop to `<MesageRow />` called `selectedLog` which should work like `selectedError` does on the errors tab ([source](https://github.com/highlight/highlight/blob/1a93b44d8aadc3929aff4fca7d81e098211b33a0/frontend/src/pages/Player/Toolbar/DevToolsWindowV2/ErrorsPage/ErrorsPage.tsx#L136-L180)).

File: frontend/src/pages/Player/Toolbar/DevToolsWindowV2/ConsolePage/ConsolePage.tsx [hunk 1]
```diff
@@ -82,46 +85,29 @@ export const ConsolePage = ({
...
-		return messagesToRender.findIndex(
-			(logEdge) => logEdge.cursor === selectedCursor,
+	const lastActiveLogIndex = useMemo(() => {
+		return findLastActiveEventIndex(
+			time,
+			sessionMetadata.startTime,
+			messageNodes,
 		)
-	}, [messagesToRender, selectedCursor])
+	}, [messageNodes, sessionMetadata.startTime, time])
```
Comment (Vadman97) [inline frontend/src/pages/Player/Toolbar/DevToolsWindowV2/ConsolePage/ConsolePage.tsx:98]: seems like this change to the deps of the useEffect below means that the `Go to` button on logs no longer works

could you see why it's not working? this is a regression from `main` as the button works fine there
Screenshot 2023-09-12 at 3 57 03 PM

**Comment by Vadman97** [comment]
how's this going @jainpawan21 ?

**Comment by rishi-raj-jain** [comment]
@jainpawan21 let me know if I can continue this and we can split.

**Comment by jainpawan21** [comment]
> @jainpawan21 let me know if I can continue this and we can split.

Hi @rishi-raj-jain, Sure you can continue this.

**Comment by jainpawan21** [comment]
> how's this going @jainpawan21 ?

Sorry Vadim for late response.
I have been caught in some work and getting very less time.

Let @rishi-raj-jain continue this 🙂

**Comment by rishi-raj-jain** [comment]
@Vadman97 are you fine with me working on this?

**Comment by Vadman97** [comment]
yes, please go ahead @rishi-raj-jain

**Comment by rishi-raj-jain** [comment]
Thanks, on it!

**Comment by rishi-raj-jain** [comment]
@vadman97  my mac broke :/

**Pull request closed without merging**